packages/video_player/ios/Classes/VideoPlayerPlugin.m

```diff
@@ -395,7 +395,7 @@ - (void)dispose {
                              forKeyPath:@"playbackBufferFull"
                                 context:playbackBufferFullContext];
   [_player replaceCurrentItemWithPlayerItem:nil];
-  [[NSNotificationCenter defaultCenter] removeObserver:self];
+  [[NSNotificationCenter defaultCenter] removeObserver:_notificationObserverId];
```

Comment:
This can be unchanged if we use the `addObserver:` method as I mentioned in the other comment.